Generated `toBuilder` signature is incorrect with generics

Given:
```java
    @Immutable
    @Value.Style(toBuilder = "toBuilder")
    interface Demo<T> {
        T getFoo();
    }
```
`immutables` generate an incorrect signature for method `toBuilder` where type parameter `<T>` would show up twice:
```java
  /**
   * Creates a builder for {@link ImmutableDemo ImmutableDemo}.prefilled
   * with attibute values of {@code this} instance to easily create modified copies.
   * @return A new ImmutableDemo builder with attributes of {@code this} instance
   */
  public ImmutableDemo.Builder<T><T> toBuilder() {
    return ImmutableDemo.<T>builder().from(this);
  }
```
Errors are:
```
[...]/ImmutableDemo.java:101: error: <identifier> expected
  public ImmutableDemo.Builder<T><T> toBuilder() {
                                 ^
[...]/ImmutableDemo.java:101: error: invalid method declaration; return type required
  public ImmutableDemo.Builder<T><T> toBuilder() {
                                     ^
```

--- a/value-fixture/src/org/immutables/fixture/builder/ToBuilderMethod.java
+++ b/value-fixture/src/org/immutables/fixture/builder/ToBuilderMethod.java
@@ -11,6 +11,13 @@ public interface ToBuilderMethod {
   }
 
   @Value.Immutable
+  @Value.Style(toBuilder = "toBuilder")
+  interface ToBuilderClassicWithGenerics<T> {
+    int a();
+    T b();
+  }
+
+  @Value.Immutable
   @Value.Style(toBuilder = "toBuilder", overshadowImplementation = true)
   interface ToBuilderSandwich {
     int a();

--- a/value-processor/src/org/immutables/value/processor/Immutables.generator
+++ b/value-processor/src/org/immutables/value/processor/Immutables.generator
@@ -207,7 +207,7 @@ public static[type.generics.def] [factoryBuildStageBuilderType type] [type.facto
    * with attibute values of {@code this} instance to easily create modified copies.
    * @return A new [type.typeValue.simple] builder with attributes of {@code this} instance
    */
-  public [type.typeBuilder.relative][type.generics.args] [type.names.toBuilder]() {
+  public [type.typeBuilder.relative] [type.names.toBuilder]() {
     return [castBuildStagedBuilder type][type.factoryBuilder.relative]()[/castBuildStagedBuilder].[type.names.from](this);
   }
 [/if]

